client/src/app/pages/source-platforms/source-platforms.tsx

```diff
@@ -0,0 +1,437 @@
+  Button,
+  ButtonVariant,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateHeader,
+  EmptyStateIcon,
+  Modal,
+  PageSection,
+  PageSectionVariants,
+  Text,
+  TextContent,
+  Toolbar,
+  ToolbarContent,
+  ToolbarGroup,
+  ToolbarItem,
+  Tooltip,
+  Popover,
+} from "@patternfly/react-core";
+import {
+  Table,
+  Tbody,
+  Th,
+  Thead,
+  Tr,
+  Td,
+  ActionsColumn,
+} from "@patternfly/react-table";
+import { CubesIcon, PencilAltIcon } from "@patternfly/react-icons";
+import { AppPlaceholder } from "@app/components/AppPlaceholder";
+import { ConditionalRender } from "@app/components/ConditionalRender";
+import { FilterToolbar, FilterType } from "@app/components/FilterToolbar";
+import { NotificationsContext } from "@app/components/NotificationsContext";
+import {
+  ConditionalTableBody,
+  TableHeaderContentWithControls,
+  TableRowContentWithControls,
+} from "@app/components/TableControls";
+import { useLocalTableControls } from "@app/hooks/table-controls";
+import {
+  useDeletePlatformMutation,
+  useFetchPlatforms,
+} from "@app/queries/platforms";
+
+import LinkToPlatformApplications from "./components/link-to-platform-applications";
+import PlatformDetailDrawer from "./components/platform-detail-drawer";
+import PlatformFormDataWaiter from "./components/platform-form";
+import { SourcePlatform, TaskState } from "@app/api/models";
+import { ConfirmDialog } from "@app/components/ConfirmDialog";
+import { getAxiosErrorMessage } from "@app/utils/utils";
+import { AxiosError } from "axios";
+import { SimplePagination } from "@app/components/SimplePagination";
+import { TablePersistenceKeyPrefix } from "@app/Constants";
+import { TaskStateIcon } from "@app/components/Icons";
+import { usePlatformProviderList } from "./usePlatformProviderList";
+
+const SourcePlatforms: React.FC = () => {
+  const { t } = useTranslation();
+  const history = useHistory();
+  const { pushNotification } = React.useContext(NotificationsContext);
+
+  const [openCreatePlatform, setOpenCreatePlatform] = useState<boolean>(false);
+
+  const [platforms, setPlatforms] = useState<SourcePlatform[] | undefined>(
+    undefined
+  );
+
+  const [platformToEdit, setPlatformToEdit] = useState<SourcePlatform | null>(
+    null
+  );
+
+  const [platformToDelete, setPlatformToDelete] =
+    React.useState<SourcePlatform | null>(null);
+
+  const {
+    platforms: basePlatforms,
+    isFetching,
+    error: fetchError,
+  } = useFetchPlatforms();
+  const baseProviderList = usePlatformProviderList();
+
+  useEffect(() => {
+    if (basePlatforms) {
+      setPlatforms(
+        basePlatforms.map((platform) => ({
+          ...platform,
+          discoverApplicationsState: "No task" as TaskState,
+        }))
+      );
+    }
+  }, [basePlatforms]);
+
+  const onError = (error: AxiosError) => {
+    pushNotification({
+      title: getAxiosErrorMessage(error),
+      variant: "danger",
+    });
+  };
+
+  const { mutate: deleteplatform } = useDeletePlatformMutation(
```

Comment:
```suggestion
  const { mutate: deletePlatform } = useDeletePlatformMutation(
```

Camel casing on variable names keeps my code spell checker happy.